lib/tasks/avalon_ualbertalib.rake

```diff
@@ -0,0 +1,71 @@
+namespace :avalon do
+    namespace :ualbertalib do
+      desc "Given a set of MediaObject IDs, in a file, one-per-line, update visibility to 'restricted'"
+      task :visibility_restricted => :environment do
+
+        options = {:commit => false}
+        option_parser = OptionParser.new do |opts|
+          opts.banner = "Usage: rake avalon:ualbertalib:visibility_restricted [options]"
+          opts.on("-c", "--commit", "Commit change; override default dry-run") { 
+            options[:commit] = true 
+          }
+          opts.on("-m ARG", "--media_objects", "File containing list of MediaObject IDs, one per line") { 
+            |filename| options[:filename] = filename 
+          }
+          opts.on("-h", "--help") do 
+             puts opts
+             exit 
+          end
+        end
+
+        # return `ARGV` with the intended arguments
+        args = option_parser.order!(ARGV) {}
+        option_parser.parse!(args)
+
+        if options[:commit] 
+          puts "Committing changes"
+        else
+          puts "Executing a 'dry-run' (no changes commited); use '-c' to save the change."
+        end
+
+        if options[:filename].nil?
+          puts "Manditory argument: -m filename"
+          puts option_parser.help
+          exit 1
+        elsif !File.exist?(options[:filename])
+          puts "Filename not found: #{options[:filename]}"
+          exit 1
+        end
+
+        mediaobject_ids = File.read(options[:filename]).split
+
+        count = 0
+        count_success = 0
+        mediaobject_ids.each do |id|
+          begin  
+            count += 1
+            obj = MediaObject.find(id)  
+            old_visibility = obj.visibility
+            obj.visibility = 'restricted' 
+            if obj.valid?
+              obj.save! if options[:commit] # default to dry-run unless commit specified 
```

Comment:
Core of the work: lookup a MediaObject, change visibility to restricted, check if valid, and save.